File: app/src/main/java/com/machinelearning/playcarddetect/modules/accessibilityaction/action/ClickAction.java
package com.machinelearning.playcarddetect.modules.accessibilityaction.action;


import android.graphics.Path;
import android.graphics.RectF;

public class ClickAction extends GestureAction {
    RectF clickRectF = new RectF();
    public ClickAction(RectF clickRectF,long startTime, long durationTime, int delayTime,String actionType) {
        super(startTime,durationTime,delayTime,actionType);
        this.clickRectF =clickRectF;
    }

    public ClickAction() {
    }

    public long getDurationTime() {
        return durationTime;
    }

    public void setDurationTime(long durationTime) {
        this.durationTime = durationTime;
    }

    public long getStartTime() {
        return startTime;
    }

    public void setStartTime(long startTime) {
        this.startTime = startTime;
    }

    public RectF getClickRectF() {
        return clickRectF;
    }

    public void setClickRectF(RectF clickRectF) {
        this.clickRectF = clickRectF;
    }

}

File: app/src/main/java/com/machinelearning/playcarddetect/modules/accessibilityaction/action/ClickAction.kt
package com.nhatran241.accessibilityactionmodule.model

import android.accessibilityservice.GestureDescription
import android.accessibilityservice.GestureDescription.StrokeDescription
import android.graphics.Path
import android.graphics.RectF
import androidx.annotation.NonNull
import androidx.annotation.Nullable
import com.machinelearning.playcarddetect.modules.accessibilityaction.action.GestureAction

class ClickAction(@NonNull rectF: RectF,@Nullable startTime:Long,@Nullable duration: Long,@Nullable delayTime : Long) : GestureAction(){
    init {
        val path = Path()
        this.delayTime =delayTime
        val builder = GestureDescription.Builder()
        path.moveTo(rectF.centerX(),rectF.centerY())
        builder.addStroke(StrokeDescription(path, startTime, duration))
        gestureDescription=builder.build()
    }
}